### PR TITLE
Add ResizeObserver sample

### PR DESCRIPTION
**Pull request by surma**
cc @atotic as an FYI

### CODE REVIEW

File: resizeobserver/index.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+<h3>Background</h3>
+<p>
+  The <a href="https://wicg.github.io/ResizeObserver/">ResizeObserver API</a>
+  provides an API to get notified whenever an element changes its size.
+</p>
+<p>
+  This sample illustrates how to use <code>ResizeObserver</code> to implement
+  media query-like behavior on a per-element basis.
```
Comment (jeffposnick) [inline resizeobserver/index.html]: Link to https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries

File: resizeobserver/demo.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/* global ResizeObserver */
+
+const ro = new ResizeObserver(entries => {
```
Comment (jeffposnick) [inline resizeobserver/demo.js:3]: This code throws a `ResizeObserver loop limit exceeded` error as soon as the page is opened. (Testing on Chrome 55.0.2883.0 with the "Experimental Web Platform Features" flag enabled.)

image

File: resizeobserver/index.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+<p>
+  The <a href="https://developers.google.com/web/updates/2016/10/resizeobserver"><em>Google Developers Web Updates</em></a>
+  article provides more details on <code>ResizeObserver</code>.
+</p>
+
+{% capture initial_output_content %}
+<p>
+  On browsers that support <code>ResizeObserver</code>, you should see a border
+  appear around the green box when resizing the window.
```
Comment (jeffposnick) [inline resizeobserver/index.html]: For folks who have difficulty distinguishing colors, could you rephrase as something like "the third box" rather than "the green box"?

Also, any particular reason why there are six boxes? It seems like you could make your point with just two boxes—one of which is associated with an RO, and the other which isn't.

File: resizeobserver/index.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  <div class="box" style="background: hsl(60, 100%, 50%)"></div>
+  <div class="box" style="background: hsl(120, 100%, 50%)"></div>
+  <div class="box" style="background: hsl(180, 100%, 50%)"></div>
+  <div class="box" style="background: hsl(240, 100%, 50%)"></div>
+  <div class="box" style="background: hsl(300, 100%, 50%)"></div>
+</div>
+{% endcapture %}
+{% include output_helper.html initial_output_content=initial_output_content %}
+
+{% capture css %}
```
Comment (jeffposnick) [inline resizeobserver/index.html]: This could just be inline in a `<style>` element alongside the other HTML. I don't think there's anything important there that developers would need to see echoed to the screen.

File: resizeobserver/index.html [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+.boxes {
+  display: flex;
+  flex-wrap: wrap;
+}
+.box {
+  height: 200px;
+  flex-basis: 200px;
```
Comment (jeffposnick) [inline resizeobserver/index.html]: Our samples template [imposes](https://github.com/GoogleChrome/samples/blob/gh-pages/styles/main.css#L7) a maximum width of `800px` on the `<body>` element when the viewport is wider than `832px`. As a result, the boxes won't change width if the window is resized until it gets below 832px.

**Review by jeffposnick — CHANGES_REQUESTED** [review]

File: resizeobserver/demo.js (hunk 2 above)
Comment (atotic) [reply resizeobserver/demo.js:3]: That's because border is being changed in observer callback, which triggers another resize event immediately.
This is not a great demonstration of RO. RO's purpose to change the Element's content in response to resize, not to change the element being resized.

File: resizeobserver/index.html (hunk 3 above)
Comment (surma) [reply resizeobserver/index.html]: Good point about the colors. The only reason it’s 6 boxes because it’s the exact same example from my /updates article. I’ll adjust accordingly

File: resizeobserver/demo.js (hunk 2 above)
Comment (surma) [reply resizeobserver/demo.js:3]: I switched to changing the background image. WDYT @atotic? This sample is more about showing usage and I wanted to keep it small. I’ll rework the sample for the `/updates` article.

screenshot 2016-10-08 09 43 22

Comment (atotic) [reply resizeobserver/demo.js:3]: This code is not buggy. But, it does not really show what RO is about. If I was a developer, my reaction would be "huh?". If we are set on two rectangles, how about making one of them a circle through border-radius, and then maintain the circleness by changing the border radius. If I saw that, I'd be like "neat!".

Comment (surma) [reply resizeobserver/demo.js:3]: I can’t quite follow. Changing the background pattern is just as much a (useless) visual change as changing the element to a circle. I am trying to show _how_ to use RO, which this easily conveys (and can be tested by the user). What am I missing?

Comment (atotic) [reply resizeobserver/demo.js:3]: It is my personal preference. These are my reasons:
- border-radius is continuous. You see circle changing smoothly every time you resize the element.
- background-image is discrete. Nothing changes unless you cross the 250px threshold. If the user is not resizing the window in the right direction, he might miss the change. Also, are we trying to make the demo phone-friendly?
- circle border that stays circle is something I'd have a very hard time coding in JS today. Seeing RO make it easy makes me a happy developer. Changing background-image based on size feels doable thought some @ rules hack.

Comment (surma) [reply resizeobserver/demo.js:3]: Ah! I now I understand. Thanks for clarifying. The continuous bit is indeed good to show off. Will change :)

**Comment by surma** [comment]
@jeffposnick I think this is good to go from our side

File: resizeobserver/index.html (hunk 4 above)
Comment (jeffposnick) [reply resizeobserver/index.html]: Now that you've moved the styles out, you should get rid of the `{% capture css %}` and corresponding `{% include ... %}`, to avoid the empty section:

image

File: resizeobserver/index.html [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+</p>
+<p>
+  This sample illustrates how to use <code>ResizeObserver</code> to implement
+  [media query](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries)
+  -like behavior on a per-element basis.
+<p>
+  The <a href="https://developers.google.com/web/updates/2016/10/resizeobserver"><em>Google Developers Web Updates</em></a>
+  article provides more details on <code>ResizeObserver</code>.
+</p>
+
```
Comment (jeffposnick) [inline resizeobserver/index.html:20]: You need a section here letting developers know that as of right now, the feature requires `chrome://flags/#enable-experimental-web-platform-features`

File: resizeobserver/index.html [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+.box {
+  height: 200px;
+  flex-basis: 200px;
+  flex-shrink: 1;
+  flex-grow: 1;
+  margin: 10px;
+  box-sizing: border-box;
+  border: 3px solid black;
+}
+.stripes {
```
Comment (jeffposnick) [inline resizeobserver/index.html]: `.stripes` isn't used anymore, right?

File: resizeobserver/index.html [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  article provides more details on <code>ResizeObserver</code>.
+</p>
+
+{% capture initial_output_content %}
+<p>
+  On browsers that support <code>ResizeObserver</code>, you should see the border
+  radius of the second box change when resizing the window.
+</p>
+<style>
+.boxes {
```
Comment (jeffposnick) [inline resizeobserver/index.html:27]: You've still got the aforementioned problem where the maximum size imposed on the `<body>` element by the default template's styles leads to a situation in which this container won't actually change size until the viewport gets below 800px or so. I.e. if your initial viewport is 1000px, and you resize the viewport to 900px, nothing happens, defeating the point of this sample.

Here's a suggestion for changing the styles so that the size of this container always takes the viewport width into account, so that any resizing will lead to the `ResizeObserver` triggering:

``` css
.boxes {
  display: flex;
  flex-wrap: wrap;
  width: 40vw;
}
.box {
  height: 200px;
  flex-shrink: 1;
  flex-grow: 1;
  margin: 10px;
  box-sizing: border-box;
  border: 3px solid black;
}
```

(It adds a `width` to `.boxes`, and removes the `flex-basis` from `.box`.)

**Review by jeffposnick — CHANGES_REQUESTED** [review]

**Comment by jeffposnick** [comment]
@surma there were a few earlier review comments that still need to be addressed (e.g. https://github.com/GoogleChrome/samples/pull/424#discussion_r82468593) and then a few new ones. Thanks!

**Comment by surma** [comment]
@jeffposnick Thanks for keeping an eye out. Apparently I was half asleep when I fixed this. Implemented all your things (hopefully).

**Review by jeffposnick — APPROVED** [review]
